schema.md

```diff
@@ -1,82 +1,71 @@
-    Does stan have access to that info? We could also just store the accepted point of a divergent trajectory.
+#### `posterior_predictive`
+Posterior predictive samples p(y|y) corresponding to the posterior predictive pdf evaluated at the `observed_data`. Samples should match with `posterior` ones and each variable should have a counterpart in `observed_data`. The `observed_data` counterpart variable may have a different name.
 
-/observed_data
-    All data that is used in observed variables (or the data (or transformed?) data sections in stan.)
+#### `constant_data`
+Model constants, data included in the model which is not modeled as a random variable (i.e. the x in a linear regression). It should be the data used to generate the `posterior` and `posterior_predictive` samples.
 
-/warnings
-    A list of warnings during sampling. Eg low effective_n, divergences....
-    TODO Not sure about the format. Can we somehow share at least part of that between stan/pymc?
-    They mostly produce the same warnings I think.
+#### `prior`
+p(theta)
 
-/prior?
-Samples from the prior distribution. Same shapes as in trace. (except for (sample, chain))
+#### `sample_stats_prior`
 
-/prior_predictive?
-Samples from the prior predictive distribution. Same vars as in /data
+#### `prior_predictive`
 
-/posterior_predictive?
-Samples from the posterior predicitve distribution. Same vars as in /data
+## Planned features
 
-/trace
-TODO We could call this /posterior
+### Sampler parameters
 
-    attrs:
-            The final parameters for the sampler. ie the final mass matrix and step size.
+### Out of sample posterior_predictive samples
+#### `predictions`
+Out of sample posterior predictive samples p(y'|y).
 
-    /trace//var1
-            One entry for each variable. The first two dimensions should always be
-            `(chain, sample)`. I guess the decision whether or not we want to expose a stacked version `draw=('chain', 'sample')`
-            is up to arviz.
-
-            Variable names must not share names with coordinate names.
-
-            attrs:
-                is_free: Whether or not the variable is a free variable for the sampler, or a transformed one
-                domain: One of (“reals”, “pos-reals”, “integers”, “sym-pos-def”, "interval"...)
-                TODO For stuff like sym-pos-def we need to know along which dims it is a matrix.
-                TODO This data could also be stored in /model
-                sym_pos_axis: [dim_idx1, dim_idx2]
-                interval_lower:
-                interval_upper
-                TODO How would this deal with cases where the lower and upper bounds depend on the index?
-
-
-
-TODO: In order to reproduce the run, it may make sense to also store some data on the random state (in numpy, this is a tuple of arrays), as
-well as some version info.  Hopefully just `PyMC3, (3, 4, 1)` or similar works.
+#### `constant_data_predictions`
```

Comment:
Minor point: suggest that we consistently name all of these attributes of `predictions` with `predictions_` as prefix.  So change this to `predictions_constant_data`.

Reply:
I like it